rllib/policy/sample_batch.py

```diff
@@ -994,34 +991,32 @@ def get_single_step_input_dict(self, view_requirements, index="last"):
 
         input_dict = {}
         for view_col, view_req in view_requirements.items():
+            if view_req.used_for_compute_actions is False:
+                continue
+
             # Create batches of size 1 (single-agent input-dict).
             data_col = view_req.data_col or view_col
             if index == "last":
                 data_col = last_mappings.get(data_col, data_col)
                 # Range needed.
                 if view_req.shift_from is not None:
-                    data = self[view_col][-1]
                     # Batch repeat value > 1: We have single frames in the
-                    # batch at each timestep.
-                    if view_req.batch_repeat_value > 1:
-                        traj_len = len(self[data_col])
-                        missing_at_end = traj_len % view_req.batch_repeat_value
-                        obs_shift = -1 if data_col in [
-                            SampleBatch.OBS, SampleBatch.NEXT_OBS
-                        ] else 0
-                        from_ = view_req.shift_from + obs_shift
-                        to_ = view_req.shift_to + obs_shift + 1
-                        if to_ == 0:
-                            to_ = None
-                        input_dict[view_col] = np.array([
-                            np.concatenate(
-                                [self[data_col][-missing_at_end:],
-                                 data])[from_:to_]
-                        ])
-                    # Batch repeat value = 1: We already have framestacks
-                    # at each timestep.
-                    else:
-                        input_dict[view_col] = data[None]
+                    # batch at each timestep (for the `data_col`).
+                    data = self[view_col][-1]
+                    traj_len = len(self[data_col])
+                    missing_at_end = traj_len % view_req.batch_repeat_value
+                    obs_shift = -1 if data_col in [
```

Comment:
can you help comment what obs_shift means?? I can't quite understand it. thanks.

Reply:
:+1:

Reply:
done